python/ray/tune/execution/ray_trial_executor.py

```diff
@@ -451,7 +451,7 @@ def _setup_remote_runner(self, trial):
             except Exception as e:
                 raise RuntimeError(
                     "Your trainable class does not accept a "
-                    "`remote_checkpoint_dir` or `custom_syncer` argument "
+                    "`remote_checkpoint_dir` or `syncer` argument "
```

Comment:
Only the `upload_dir` is, which is missing the experiment name + trial dir name.

`remote_checkpoint_dir = {sync_config.upload_dir}/{run_config.name}/{trial.relative_logdir}`

I think this could be made a bit cleaner if we consolidate local_dir/upload_dir/name like we discussed before!